packages/core/src/clients/create-account-public-client.ts

```diff
@@ -0,0 +1,38 @@
+import { CosmWasmClient } from '@cosmjs/cosmwasm-stargate'
+import { VersionControlTypes } from 'src/codegen/abstract/index.js'
+import type { Prettify } from '../types/utils.js'
+import {
+  type Client,
+  type ClientConfig,
+  createClient,
+} from './create-client.js'
+import {
+  type AccountPublicActions,
+  accountPublicActions,
+} from './decorators/account-public.js'
+
+export type AccountPublicClientConfig = ClientConfig & {
+  accountId: VersionControlTypes.AccountId
```

Comment:
`chainName` is "inferrable" from the `CosmWasmClient` chain-id, but IMO would be better to provide the full `AccountId` type with the chain name for sanity checking that we're getting the account on the correct chain.

Reply:
Yes, but there are cases (like in the account creation page on the Console) where multiple CosmWasm clients are in use, which could have the dev pass in an incorrect one. Best practice would be, however, to check the chain id of the cosmwasm client, of course.

Something that's missing in the AccountId logic as it stands is that the FE enforced that "local" account ids were never allowed because they're ambiguous, and now we're allowing them again.